lib/actions/device.coffee

```diff
@@ -128,25 +128,40 @@ exports.register =
 		Examples:
 
 			$ resin device register MyApp
+			$ resin device register MyApp --uuid 83cb718aee3abe63d2e0a0016ea49399cdc5b3f9b879dbb01acf73f28f5085
+			$ resin device register MyApp --uuid 83cb718aee3abe63d2e0a0016ea49399cdc5b3f9b879dbb01acf73f28f5085 --device-api-key existingDeviceKey
 	'''
 	permission: 'user'
 	options: [
-		signature: 'uuid'
-		description: 'custom uuid'
-		parameter: 'uuid'
-		alias: 'u'
+		{
+			signature: 'uuid'
+			description: 'custom uuid'
+			parameter: 'uuid'
+			alias: 'u'
+		}
+		{
+			signature: 'deviceApiKey'
+			description: 'custom device key'
+			parameter: 'device-api-key'
+			alias: 'k'
+		}
 	]
 	action: (params, options, done) ->
 		Promise = require('bluebird')
 		resin = require('resin-sdk-preconfigured')
 
-		resin.models.application.get(params.application).then (application) ->
-
-			Promise.try ->
-				return options.uuid or resin.models.device.generateUniqueKey()
-			.then (uuid) ->
+		Promise.join(
+			resin.models.application.get(params.application)
+			options.uuid or resin.models.device.generateUniqueKey()
+			options.deviceApiKey ? resin.models.device.generateUniqueKey()
```

Comment:
We use `or` in the previous line and `?` here for the same purpose, so maybe let's make them consistent?